api/src/main/java/org/eclipse/microprofile/health/spi/SPIFactory.java

```diff
@@ -19,14 +19,16 @@
  * SPDX-License-Identifier: Apache-2.0
  *
  */
-package org.eclipse.microprofile.health;
+
+package org.eclipse.microprofile.health.spi;
+
+import org.eclipse.microprofile.health.ResponseBuilder;
 
 /**
- * The basic health check procedure interface
- *
- * @author Heiko Braun
- * @since 13.06.17
+ * Created by hbraun on 07.07.17.
  */
-public interface HealthCheckProcedure {
-    HealthStatus perform();
+public interface SPIFactory {
+
+    ResponseBuilder createResponseBuilder();
+
```

Comment:
I think the structure is confusing. You need Response, Response Builder and ResponseResolver. ResponseResolver is to use service loader pattern to load an implementation of ResponseResolver. Response should just be a clean object to contain name, attributes, up, down, state etc.